### PR TITLE
Dependency update

### PR DESCRIPTION
**Pull request by nikolaevtomov**
- Update dependencies
- Swap babel-eslint for @babel/eslint-parser https://github.com/dabapps/eslint-config-dabapps/issues/76

Note: I intended to publish this as `major` `v8.0.0`.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -62,6 +62,6 @@
     "webpack": "^4.41.6"
   },
   "peerDependencies": {
-    "eslint": "^6.8.0"
+    "eslint": "^7.21.0"
```
Comment (JakeSidSmith) [inline package.json:65]: This is a major version, therefore we bump a major version.

File: package.json [hunk 2]
```diff
@@ -35,18 +35,18 @@
   },
   "homepage": "https://github.com/dabapps/eslint-config-dabapps",
   "dependencies": {
-    "@typescript-eslint/eslint-plugin": "^4.15.2",
-    "@typescript-eslint/parser": "^4.15.2",
```
Comment (JakeSidSmith) [inline package.json:39]: Why have you uninstalled the typescript parser?

Comment (nikolaevtomov) [reply package.json:39]: Probably forgot to put it back, reverted 👍

**Review by JakeSidSmith — APPROVED** [review]
Happy to go to version `8.0.0`

**Pull request merged**